test/e2e/serial/config/config.go

```diff
@@ -67,3 +75,23 @@ func GetRteCiImage() string {
 	}
 	return nropTestCIImage
 }
+
+func CheckNodesTopology() error {
+	var nrtList nrtv1alpha2.NodeResourceTopologyList
+
+	err := e2eclient.Client.List(context.TODO(), &nrtList)
+	if err != nil {
+		return fmt.Errorf("error while trying to get NodeResourceTopology objects. error: %w", err)
+	}
+	// we're ok with any TM policy as long as the updater can handle it,
+	// we use this as proxy for "there is valid NRT data for at least X nodes
+	policies := []nrtv1alpha2.TopologyManagerPolicy{
+		nrtv1alpha2.SingleNUMANodeContainerLevel,
+		nrtv1alpha2.SingleNUMANodePodLevel,
+	}
+	nrts := e2enrt.FilterByPolicies(nrtList.Items, policies)
+	if len(nrts) < minNumberOfNodesWithRightTopology {
```

Comment:
this is exactly the case we want to avoid. If there are 2 nodes with different policies there must be something wrong. NRT data are expected to align with kubeletconfig settings, if it's not then there is no point of running the tests thus skipping the suite. 
what I'd do is to fetch the kubelet configuration of numa topology and scope policy and compare it to what's there in NRT, so the check is correct but the condition on the policies is not.

Reply:
right. 
the reason we need this change is that in case the cluster was corrupted for some reason (bad configuration, mcp misconfiguration, tests leftovers.. ) the tests continued being executed and then getting skipped later because when they checked for the minimum nodes with matching tmpolicy there weren't enough nodes. so via this modification, we aim to fail fast and skip the suite even earlier.